src/basics.md

```diff
@@ -470,6 +471,48 @@ fn main() {
 }
 ```
 
+[ex-phone]: #ex-phone
+<a name="ex-phone"></a>
+## Extract phone numbers from text
+
+Attempts to process a user input as a US phone number.  Essentially, the
+expression is checking that there are 7 digits that come in groups of 3 3 4,
+removing any extra characters.  The phone number is then formatted for display.
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate regex;
+
+use std::io::stdin;
+use regex::Regex;
+#
+# error_chain!{
+#     foreign_links {
+#         Regex(regex::Error);
+#         Io(std::io::Error);
+#     }
+# }
+
+fn run() -> Result<()> {
+    println!("Input a phone number");
+    let mut phone_input = String::new();
+    // Regular expression is looking for 3 groups of digits, capturing them
+    let re = Regex::new(r".*([2-9]\d{2}).*(\d{3}).*(\d{4})")?;
+    stdin().read_line(&mut phone_input)?;
```

Comment:
Good ideas, but should we just use a static string so that it works in a cookbook?